rust/src/tx_builder.rs

```diff
@@ -399,10 +456,117 @@ impl TransactionBuilder {
+        let mut metadata = self.auxiliary_data.as_ref()
+            .map(|aux| { aux.metadata().as_ref().cloned() })
+            .unwrap_or(None)
+            .unwrap_or(GeneralTransactionMetadata::new());
+        metadata.insert(key, val);
+        self.set_metadata(&metadata);
+    }
+
+    /// Add a single JSON metadatum using a TransactionMetadatumLabel and a String
+    /// It will be securely added to existing or new metadata in this builder
+    pub fn add_json_metadatum(
+        &mut self,
+        key: &TransactionMetadatumLabel,
+        val: String,
+    ) -> Result<(), JsError> {
+        self.add_json_metadatum_with_schema(key, val, MetadataJsonSchema::NoConversions)
+    }
+
+    /// Add a single JSON metadatum using a TransactionMetadatumLabel, a String, and a MetadataJsonSchema object
+    /// It will be securely added to existing or new metadata in this builder
+    pub fn add_json_metadatum_with_schema(
+        &mut self,
+        key: &TransactionMetadatumLabel,
+        val: String,
+        schema: MetadataJsonSchema,
+    ) -> Result<(), JsError> {
+        let metadatum = encode_json_str_to_metadatum(val, schema)?;
+        self.add_metadatum(key, &metadatum);
+        Ok(())
+    }
+
+    /// Set explicit Mint object to this builder
+    /// it will replace any previously existing mint
+    pub fn set_mint(&mut self, mint: &Mint) {
+        self.mint = Some(mint.clone());
+    }
+
+    /// Add a mint entry to this builder using a PolicyID and MintAssets object
+    /// It will be securely added to existing or new Mint in this builder
+    /// It will replace any existing mint assets with the same PolicyID
+    pub fn set_mint_asset(&mut self, policy_id: &PolicyID, mint_assets: &MintAssets) {
+        let mut mint = self.mint.as_ref().cloned().unwrap_or(Mint::new());
+        mint.insert(policy_id, mint_assets);
+        self.set_mint(&mint);
+    }
+
+    /// Add a mint entry to this builder using a PolicyID, AssetName, and Int object for amount
+    /// It will be securely added to existing or new Mint in this builder
+    /// It will replace any previous existing amount same PolicyID and AssetName
+    pub fn add_mint_asset(&mut self, policy_id: &PolicyID, asset_name: &AssetName, amount: Int) {
+        let mut asset = self.mint.as_ref()
+            .map(|m| { m.get(policy_id).as_ref().cloned() })
+            .unwrap_or(None)
+            .unwrap_or(MintAssets::new());
+        asset.insert(asset_name, amount);
+        self.set_mint_asset(policy_id, &asset);
+    }
+
+    /// Add a mint entry together with an output to this builder
+    /// Using a PolicyID, AssetName, Int for amount, Address, and Coin (BigNum) objects
+    /// The asset will be securely added to existing or new Mint in this builder
+    /// A new output will be added with the specified Address, the Coin value, and the minted asset
+    pub fn add_mint_asset_and_output(
```

Comment:
Do we really need to combine these two operations? We're already adding a **lot** of convenience functions here.

Reply:
Don't see a reason not to have it, we want to make tx-builder easier and quicker to use for dapps and the prediction is that 90% of case when something is minted it will be just sent to one receiver right away, so it's a good way to save a buncha boilerplate code especially with all the different types between the mint assets and output assets. It's not any bigger complication to test or support it.

Good thing that you pointed me to these functions cuz I was thinking to delete them at first but then noticed that they are actually implemented incorrectly and now they are fixed and they are encapsulating a bit more complicated code that only adds the single specified entry to the output and not the entire mint in the builder.